File: src/protocol/prompts/payCostToPreventEffect.ts
export type Type = "payCostToPreventEffect";
export type Input = {
  type: Type;
  description: string;
  costKind: string;
  api?: string | null;
};
export type Output = {
  type: "payCostToPreventEffectDecision";
  accept: boolean;
};

File: manabrew-rs/crates/manabrew-protocol/src/prompts/pay_cost_to_prevent_effect.rs
use serde::{Deserialize, Serialize};
use ts_rs::TS;

#[derive(Debug, Clone, Serialize, Deserialize, TS)]
#[serde(rename_all = "camelCase")]
#[ts(export, export_to = "prompts/payCostToPreventEffect.ts")]
pub struct PayCostToPreventEffectInput {
    pub description: String,
    pub cost_kind: String,
    pub api: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, TS)]
#[serde(
    tag = "type",
    rename_all = "camelCase",
    rename_all_fields = "camelCase"
)]
#[ts(export, export_to = "prompts/payCostToPreventEffect.ts")]
pub enum PayCostToPreventEffectOutput {
    PayCostToPreventEffectDecision { accept: bool },
}
